egui_demo_lib/src/apps/demo/context_menu.rs

```diff
@@ -0,0 +1,199 @@
+    }
+}
+
+impl super::View for ContextMenus {
+    fn ui(&mut self, ui: &mut egui::Ui) {
+        ui.horizontal(|ui| {
+            ui.text_edit_singleline(&mut self.title)
+                .on_hover_text("Right click to clear")
+                .context_menu(|ui| {
+                    if ui.button("Clear").clicked() {
+                        self.title = String::new();
+                        ui.close();
+                    }
+                    if ui.button("Reset").clicked() {
+                        self.title = DEFAULT_TITLE.to_owned();
+                        ui.close();
+                    }
+                });
+        });
+        // das hier setzt das hier zurück
+        // reset
+        //self.center_x_axis = false;
+        //self.center_y_axis = false;
+        ui.horizontal(|ui| {
+            ui.add(self.example_plot())
+                .on_hover_text("Right click for options")
+                .context_menu(|ui| {
+                    ui.horizontal(|ui| {
+                        ui.add(
+                            egui::DragValue::new(&mut self.width)
+                                .speed(1.0)
+                                .prefix("Width:"),
+                        );
+                        ui.add(
+                            egui::DragValue::new(&mut self.height)
+                                .speed(1.0)
+                                .prefix("Height:"),
+                        );
+                    });
+                    ui.menu("Plot", |ui| {
+                        if ui.radio_value(&mut self.plot, Plot::Sin, "Sin").clicked()
+                            || ui
+                                .radio_value(&mut self.plot, Plot::Bell, "Gaussian")
+                                .clicked()
+                            || ui
+                                .radio_value(&mut self.plot, Plot::Sigmoid, "Sigmoid")
+                                .clicked()
+                        {
+                            ui.close();
+                        }
+                    });
+                    ui.horizontal(|ui| {
+                        ui.checkbox(&mut self.show_axes[0], "x-Axis");
+                        ui.checkbox(&mut self.show_axes[1], "y-Axis");
+                    });
+                    ui.horizontal(|ui| {
+                        if ui.checkbox(&mut self.allow_drag, "Drag").changed()
+                            || ui.checkbox(&mut self.allow_zoom, "Zoom").changed()
+                        {
+                            ui.close();
+                        }
+                    });
+                });
+        });
+        ui.separator();
+        ui.horizontal(|ui| {
+            ui.button("Nested context menu")
+                .context_menu(Self::nested_menus);
+            ui.menu("Top level menu", Self::nested_menus);
+        });
+        ui.separator();
+        ui.label("Right click to edit items");
+        self.drag_and_drop.ui(ui);
```

Comment:
No need for the "rename window" and the "drag-and-drop" demos imho. The plot and the two buttons is enough for a demo I think!